### DISCUSSION TITLE
transitionEnd or animationEnd on sprite component

### DISCUSSION THREAD

**Discussion by ryanmunger** [replit/kaboom#218] [Q&A] [answered]
Hello! Anyway to tap into using transitionEnd or animationEnd in regards to the sprite component’s play method in order to queue up other actions after an animation ends. It would be nice to even chain animations together the do something else (call function, run code whatever) after it’s all done. Or is there a similar implementation I can use?

Thanks!

Ryan

**Comment by slmjkdbtl** [comment, marked as answer]
There's an `"animEnd"` event fired when an animation ends, I'm also adding an `onEnd()` callback to `play()`

```js
player.on("animEnd", (anim) => {
    if (anim === "jump") {
        // ...
    }
});
```

**Reply by ryanmunger** [reply]
@slmjkdbtl thank you so much for the quick response. Okay - one other quick question if you don’t mind! Is there also a way to fire an event after the move() or a position change of a sprite has been called? I am working on a turn based battler like final fantasy 1 and wanted to move the character forward a few px, do an animation (sword swipe, whatever), then move back to starting position then move to the next battlers turn after movement/position change, animation, dmg is dealt, etc.

you do an amazing job maintaining this project and keeping up with the community by the way!

Ryan

**Reply by ryanmunger** [reply]
I was even thinking maybe a trigger in my own custom component in my codebase.

add([sprite(), battler()])

battler could have the sprite move, animations fire, then call this.trigger(‘turnEnd’) or something. I will post if I can come up with something cool like that :)

**Reply by slmjkdbtl** [reply]
Yeah! Make some custom components. There's no move event like that, but you can definitely encapsulate that into something like 
```js
battle() {
    const origPos = this.pos;
    this.pos = this.pos.add(-20, 20);
    wait(0.5, () => this.pos = origPos);
    this.trigger("battle");
}
```

**Reply by ryanmunger** [reply]
Great! I will try some things like that. Thank you again @slmjkdbtl!

✌️
